lib/instrument.ex

```diff
@@ -299,64 +299,167 @@ defmodule PropCheck.Instrument do
+  # of shared state (in particular for :ets, Registry and the process dictionary)
+  all_instrumentable_functions = [
+    ets: :lookup,
+    ets: :lookup_element,
+    ets: :update_element,
+    ets: :insert,
+    ets: :insert_new,
+    ets: :delete_object,
+    ets: :delete,
+    ets: :delete_all_objects,
+    ets: :select_delete,
+    ets: :match_delete,
+    ets: :match_object,
+    ets: :member,
+    ets: :new,
+
+    gen_server: :start_link,
+    gen_server: :start,
+    gen_server: :call,
+    gen_server: :cast,
+    gen_server: :server,
+    gen_server: :loop,
+
+    supervisor: :start_link,
+    supervisor: :start_child,
+    supervisor: :which_children,
+
+    timer: :sleep,
+    timer: :apply_after,
+    timer: :exit_after,
+
+    erlang: :spawn,
+    erlang: :spawn_link,
+    erlang: :link,
+    erlang: :process_flag,
+    # erlang: yield,
+    erlang: :now,
+    erlang: :is_process_alive,
+    erlang: :demonitor,
+    erlang: :monitor,
+    erlang: :exit,
+
+    gen_event: :start_link,
+    gen_event: :send,
+    gen_event: :add_handler,
+    gen_event: :notify,
+
+    gen_fsm: :start_link,
+    gen_fsm: :send_event,
+    gen_fsm: :send_all_state_event,
+    gen_fsm: :sync_send_all_state_event,
+
+    io: :format,
+
+    file: :write_file,
+
+    IO: :puts,
+
+    GenServer: :start_link,
+    GenServer: :start,
+    GenServer: :call,
+    GenServer: :cast,
+    GenServer: :server,
+    GenServer: :loop,
+
+    Task: :start_link,
+    Task: :start,
+    Task: :call,
+    Task: :cast,
+    Task: :shutdown,
+    Task: :async,
+    Task: :await,
+    Task: :async_stream,
+    Task: :yield,
+    Task: :yield_many,
+
+    Supervisor: :start_link,
+    Supervisor: :start_child,
+    Supervisor: :restart_child,
+    Supervisor: :stop,
+    Supervisor: :count_children,
+    Supervisor: :delete_child,
+    Supervisor: :terminate_child,
+    Supervisor: :which_children,
+
+    DynamicSupervisor: :start_link,
+    DynamicSupervisor: :start_child,
+    DynamicSupervisor: :stop,
+    DynamicSupervisor: :count_children,
+    DynamicSupervisor: :delete_child,
+    DynamicSupervisor: :terminate_child,
+    DynamicSupervisor: :which_children,
+
+    Agent: :start_link,
+    Agent: :start,
+    Agent: :cast,
+    Agent: :stop,
+    Agent: :get,
+    Agent: :get_and_update,
+    Agent: :update,
+
+    Process: :delete_key,
+    Process: :get,
+    Process: :link,
+    Process: :put,
+    Process: :register,
+    Process: :whereis,
+
+    Registry: :start_link,
+    Registry: :count_match,
+    Registry: :count,
+    Registry: :dispatch,
+    Registry: :keys,
+    Registry: :lookup,
+    Registry: :match,
+    Registry: :meta,
+    Registry: :put_meta,
+    Registry: :register,
+    Registry: :select,
+    Registry: :unregister_match,
+    Registry: :unregister,
+    Registry: :update_value,
+  ] ++
+  [ # the name with the dot requires a different encoding
+    {:"Task.Supervisor", :start_link},
```

Comment:
It should be possible to write `"Task.Supervisor": :start_link`.

Reply:
Silly me! yes, this is the correct syntax, which I never got right!